src/application/action/main.ts

```diff
@@ -0,0 +1,139 @@
+import fs from 'node:fs';
+import os from 'node:os';
+
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+import path from 'path';
+
+interface Inputs {
+  githubRepo: string;
```

Comment:
nit: should the non-required ones have nullable types?

Reply:
The GitHub input parser `core.getInput` will return an empty string when an input isn't supplied. I could make those types nullable, but then I'd have to do the extra work of making them null if they are empty :/.